codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameWriter.java

```diff
@@ -56,9 +56,10 @@
      * @param endStream indicates if this is the last frame to be sent for the stream.
      * @param promise the promise for the write.
      * @return the future for the write.
+     * @throws Http2Exception if an exception occurs while encoding headers.
      */
     ChannelFuture writeHeaders(ChannelHandlerContext ctx, int streamId, Http2Headers headers,
-            int padding, boolean endStream, ChannelPromise promise);
+            int padding, boolean endStream, ChannelPromise promise) throws Http2Exception;
```

Comment:
My primary concern here is that the docs give users overriding this functionality some breadcrumbs to make sure they don't break their HTTP/2 stack.  Doing the wrong thing when overriding, as we have discovered, can lead to header table corruption, and potentially cross stream contamination.